org.eclipse.richbeans.widgets/src/org/eclipse/richbeans/widgets/shuffle/ShuffleViewer.java

```diff
@@ -0,0 +1,258 @@
+				aTable.setSelection(new StructuredSelection(rem.get(aindex)));
+			}
+		} catch (Exception ne) {
+			logger.error("Cannot set the values after move!", ne);
+		}
+	}
+	
+	
+	private Object getSelection(TableViewer viewer) {
+		ISelection sel = viewer.getSelection();
+		if (sel instanceof StructuredSelection) {
+			return ((StructuredSelection)sel).getFirstElement();
+		}
+		return null;
+	}
+
+
+	private final TableViewer createTable(Composite parent, 
+			                              String tooltip, 
+			                              List<Object> items, 
+			                              String propName,
+			                              String slabel,
+			                              boolean selectFromEnd, 
+			                              boolean allowReorder) {
+		
+		Composite content = new Composite(parent, SWT.NONE);
+		content.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, true));	
+		content.setLayout(new GridLayout(1, false));
+		content.setBackground(content.getDisplay().getSystemColor(SWT.COLOR_WHITE));
+		
+		if (slabel!=null) {
+			final Label label = new Label(content, SWT.NONE);
+			label.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, false));	
+			label.setText(slabel);
+			label.setBackground(content.getDisplay().getSystemColor(SWT.COLOR_WHITE));
+		}
+				
+		TableViewer ret = new TableViewer(content, SWT.SINGLE | SWT.H_SCROLL | SWT.V_SCROLL | SWT.BORDER | SWT.FULL_SELECTION);
+		ret.getControl().setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, true));	
+		((Table)ret.getControl()).setToolTipText(tooltip); // NOTE This can get clobbered if we used tooltips inside the table.
+		
+		final ShuffleContentProvider prov = new ShuffleContentProvider(conf, propName, selectFromEnd);
+		ret.setContentProvider(prov);
+		
+		ret.setInput(items);
+		
+		Composite buttons = new Composite(content, SWT.NONE);
+		buttons.setBackground(content.getDisplay().getSystemColor(SWT.COLOR_WHITE));
+		buttons.setLayoutData(new GridData(SWT.CENTER, SWT.FILL, true, false));
+		buttons.setLayout(new GridLayout(2, true));
+		GridUtils.removeMargins(buttons, 20, 0);
+		
+		Button down = new Button(buttons, SWT.ARROW | SWT.DOWN);
+		down.setBackground(content.getDisplay().getSystemColor(SWT.COLOR_WHITE));
+		down.setEnabled(false);
+		GridData gd = new GridData(SWT.FILL, SWT.FILL, true, true);
+		gd.widthHint = 50;
+		down.setLayoutData(gd);
+		down.addSelectionListener(new SelectionAdapter() {
+        	public void widgetSelected(SelectionEvent e) {
+        		moveVertical(ret, propName, 1);
+        	}
+        });
+		
+		Button up = new Button(buttons, SWT.ARROW | SWT.UP);
+		up.setBackground(content.getDisplay().getSystemColor(SWT.COLOR_WHITE));
+		up.setEnabled(false);
+		up.setLayoutData(gd);	
+		up.addSelectionListener(new SelectionAdapter() {
+        	public void widgetSelected(SelectionEvent e) {
+        		moveVertical(ret, propName, -1);
+        	}
+        });
+		if (!allowReorder) { // This cannot be changed later
```

Comment:
Although it's not quite the same thing in this case because I start with them visible.